### DISCUSSION TITLE
query online HDT file

### DISCUSSION THREAD

**Discussion by mathib** [comunica/comunica#498] [Q&A] [answered]
#### Issue type:

- :question: Question

____
#### Question:

Hi, nice work with this library! 

I'm just exploring the current options of Comunica and I wondered if it is possible to query over remote HDT files using `actor-init-sparql`, similarly as can be done with remote turtle files. If I test this I get a vague error: `punycode.js:2 Uncaught (in promise) Error`.

**Comment by rubensworks** [comment, marked as answer]
Querying remote HDT files is not supported (at the moment) unfortunately.

However, we do have [a separate package](https://github.com/comunica/comunica-actor-init-sparql-hdt) to query local HDT files. So you could use this, assuming that you'd download the remote HDT files before querying.

**Comment by mathib** [comment]
Thanks for the update. I saw the separate package afterwards. Querying remote HDT files seems a (really) 'nice to have'. Is this functionality somewhere on the roadmap?

**Comment by rubensworks** [comment]
The problem is that the HDT format is not really supposed to be queried directly over the Web, it is mainly to be used as an internal index for query engines. However, the [TPF interface](https://linkeddatafragments.org/) (which is queryable by Comunica) can be setup with an HDT backend, which can be queried very efficiently over HTTP(S).

So for this reason, querying remote HDT files is not on the roadmap, as this would require you to download (possibly huge) files every time you want to query one.

**Comment by mathib** [comment]
Fair point :) I didn't know that LDF-server offered TPF with HDT backend. Thanks for the pointer!
